ls1intum/Artemis
# `Exam mode`: Update submitted and started exam details in real time

### Checklist
#### General


#### Client

### Motivation and Context



@krusche asked me to implement this functionality as my final task. This will allow instructors to stay on the exam configuration page without reloading.

### Description


- We send the following messages once a student submits (`/topic/exam/${exam.id}/submitted`) or starts (`/topic/exam/${exam.id}/started`) his exam

### Steps for Testing

Prerequisites:
- 1 Instructor
- Multiple Students
- Exam

1. Log in to Artemis
2. Navigate to Course Administration
3. Create an Exam
4. Register multiple students
5. Participate with those students and keep the `Exam Management` page open (see screenshots)
6. You should notice that the statistics are updated (e.g. started and submitted student)
7. This also happens for the exam checklist (see screenshots)

### Review Progress





#### Code Review
#### Manual Tests

Tests unchanged and won`t be added in this PR.

### Screenshots



<details>
<summary>Testing Screenshots</summary>

![image](https://user-images.githubusercontent.com/60005702/185210396-3735d73c-f364-49b3-a6e6-de984832bfd6.png)
![image](https://user-images.githubusercontent.com/60005702/185210486-edebbd6a-106f-467f-927c-31548e278c71.png)
</details>
